### ISSUE TITLE
Trivial default.nix setup times out on Binder

### ISSUE THREAD

**Issue by l0b0** [jupyterhub/repo2docker#1143]
## Bug description

The following `default.nix` setup times out [on Binder](https://mybinder.org/v2/gl/victor-engmark%2Fbash-scripting-intro/HEAD?labpath=intro.ipynb):

```nix
{ pkgs ? import ./nixpkgs.nix
}:
(
  pkgs.python3.withPackages (
    ps: [
      ps.bash_kernel
      ps.jupyterlab
    ]
  )
)
```

#### Expected behaviour

Should work as well as `nix-shell --pure --run 'jupyter lab intro.ipynb'` works locally.

#### Actual behaviour

Times out after a few minutes; see [logs](https://gist.github.com/l0b0/ba9a1b1938fc7e25ecc66ed8de430352).

### How to reproduce

1. Go to https://mybinder.org/v2/gl/victor-engmark%2Fbash-scripting-intro/HEAD?labpath=intro.ipynb
2. Wait for it to time out

### Your personal set up

Not applicable, and I don't know what the values are on Binder.

**Comment by meeseeksmachine** [comment]
This issue has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/trivial-nix-based-notebook-fails-to-load/13172/5

**Comment by robertodr** [comment]
This is still an issue. Not even the examples [repository](https://github.com/binder-examples/nix) works.

**Comment by jezcope** [comment]
Yup, I've run into this when trying some things out thinking how #1300 might be implemented.

The weird thing is that running `repo2docker` locally for a nix-based repository works fine so I guess it's something to do with how JupyterHub detects that the container is ready?
